src/commands/moderation/prefixCommands/functions/setContent.ts

```diff
@@ -0,0 +1,232 @@
+        .setCustomId('commandContentContent')
+        .setLabel('Content')
+        .setPlaceholder('Provide the content for the command.')
+        .setStyle(TextInputStyle.Paragraph)
+        .setMaxLength(2048)
+        .setMinLength(0)
+        .setRequired(false)
+        .setValue(foundContent && foundContent.content ? foundContent.content : '');
+
+    const commandContentImageUrl = new TextInputBuilder()
+        .setCustomId('commandContentImageUrl')
+        .setLabel('Image URL')
+        .setPlaceholder('Provide an optional Image URL for the command.')
+        .setStyle(TextInputStyle.Short)
+        .setMaxLength(255)
+        .setMinLength(0)
+        .setRequired(false)
+        .setValue(foundContent && foundContent.image ? foundContent.image : '');
+
+    const titleActionRow = new ActionRowBuilder<TextInputBuilder>().addComponents(commandContentTitle);
+    const contentActionRow = new ActionRowBuilder<TextInputBuilder>().addComponents(commandContentContent);
+    const imageUrlActionRow = new ActionRowBuilder<TextInputBuilder>().addComponents(commandContentImageUrl);
+
+    contentModal.addComponents(titleActionRow);
+    contentModal.addComponents(contentActionRow);
+    contentModal.addComponents(imageUrlActionRow);
+
+    await interaction.showModal(contentModal);
+
+    const filter = (interaction: {
+        customId: string;
+        user: { id: any; };
+    }) => interaction.customId === 'commandContentModal' && interaction.user.id;
+
+    let title = '';
+    let content = '';
+    let image = '';
+
+    try {
+        //Await a modal response
+        const modalSubmitInteraction = await interaction.awaitModalSubmit({
+            filter,
+            time: 120000,
+        });
+
+        title = modalSubmitInteraction.fields.getTextInputValue('commandContentTitle').trim();
+        content = modalSubmitInteraction.fields.getTextInputValue('commandContentContent').trim();
+        image = modalSubmitInteraction.fields.getTextInputValue('commandContentImageUrl').trim();
+
+        if (!title && !content && !image) {
+            await modalSubmitInteraction.reply({
+                content: 'You did not provide any content information and the change was not made.',
+                ephemeral: true,
+            });
+            return;
+        }
+        await modalSubmitInteraction.reply({
+            content: 'Processing command content data.',
+            ephemeral: true,
+        });
+    } catch (error) {
+        //Handle the error if the user does not respond in time
+        Logger.error(error);
+        await interaction.followUp({
+            content: 'You did not provide the necessary content information and the change was not made.',
+            ephemeral: true,
+        });
+        return;
+    }
+    //Check if the mod logs channel exists
+    const modLogsChannel = interaction.guild.channels.resolve(constantsConfig.channels.MOD_LOGS) as TextChannel;
+    if (!modLogsChannel) {
```

Comment:
```suggestion
    let modLogsChannel = interaction.guild.channels.resolve(constantsConfig.channels.MOD_LOGS);
    if (!modLogsChannel || !modLogsChannel.isTextBased()) {
        modLogsChannel = null;
```
Might be safer to actually check if the channel is a text channel? Especially if we want to offer the codebase to other projects. Then, if it's not a text channel set it to null to prevent accidental attempts to send into a non-text channel.